tests/test_VNET.py

```diff
@@ -142,3 +142,25 @@ def test_vnet_mac_address_can_be_configured(
             f"vnet7:{existing_jail.jid}"
         ]).decode("utf-8")
         assert mac_a in stdout
+
+    def test_devfs_ruleset_for_dhcp(
+        self,
+        existing_jail: 'libioc.Jail.Jail',
+        bridge_interface: str
+    ) -> None:
+        existing_jail.config["devfs_ruleset"] = 4
+        existing_jail.config["vnet"] = True
+        existing_jail.config["interfaces"] = f"vnet0:{bridge_interface}"
+
+        assert existing_jail.devfs_ruleset == 4
+        existing_jail.config["ip4_addr"] = "vnet0|dhcp"
+
+        assert existing_jail.devfs_ruleset > 4
+        stdout = subprocess.check_output([
+            "/sbin/devfs",
+            "rule",
+            "-s",
+            str(existing_jail.devfs_ruleset),
+            "show"
+        ]).decode("utf-8")
+        assert "bpf" in stdout
```

Comment:
could you also check that the process dhclient is there?

Reply:
The jail is not started during this test because there is no DHCP Server that can handle the request. The latest commit starts the jail and checks for the availability of the BPF device.

Indeed, it would be interesting to know if the jail can query an IP address with dhclient, but with this test we have now ensured that it's technically possible.